optuna/samplers/_tpe/sampler.py

```diff
@@ -622,7 +622,7 @@ def _get_observation_pairs(
     scores = []
     values: Dict[str, List[Optional[float]]] = {param_name: [] for param_name in param_names}
     violations: Optional[List[float]] = [] if constraints_enabled else None
-    for trial in study.get_trials(deepcopy=False, states=states):
+    for trial in study._get_trials(deepcopy=False, states=states, use_cache=True):
```

Comment:
Does it sound reasonable to make `use_cache=False` when `constant_liar` option is true?

Reply:
This is a good point. I discussed this with @c-bata -san in person and concluded that we should do this.
When `constant_liar` is `True`, parameters of latest trials is necessary to avoid evaluating the similar parameters in distributed environment.